onnx/defs/generator/defs.cc

```diff
@@ -1,227 +1,292 @@
+    .SetDoc(R"DOC(
+Generate a tensor with random values drawn from a normal distribution. 
+The shape of the output tensor is copied from the shape of the input tensor, 
+and the parameters of the normal distribution are specified by `mean` and `scale`.
+
+The data type is specified by the 'dtype' argument, or copied from the input tensor if not provided. 
+The 'dtype' argument must be one of the data types specified in the 'DataType' enum field in the
+TensorProto message, and be valid as an output type.
+)DOC")
+    .Attr(
+          "mean",
+          "The mean of the normal distribution. If not specified, default is 0.",
+          AttributeProto::FLOAT,
+          0.0f)
+    .Attr(
+          "scale",
+          "The standard deviation of the normal distribution. If not specified, default is 1.",
+          AttributeProto::FLOAT,
+          1.0f)
+    .Attr(
+          "seed",
+          "(Optional) Seed to the random generator, if not specified we will auto generate one.",
+          AttributeProto::FLOAT,
+          OPTIONAL)
+    .Attr(
+          "dtype",
+          "(Optional) The data type for the elements of the output tensor, if not specified, we will use"
+          "the data type of the input tensor.",
+          AttributeProto::INT,
+          OPTIONAL)
+    .Input(
+           0,
+           "input",
+           "Input tensor to copy shape and optionally type information from.", "T1")
+    .Output(
+            0,
+            "output",
+            "Output tensor of random values drawn from normal distribution", "T2")
+    .TypeConstraint(
+        "T1",
+        OpSchema::all_tensor_types(),
+        "Constrain to any tensor type. If the dtype attribute is not provided this must be a valid output type.")
+    .TypeConstraint("T2", { "tensor(float16)", "tensor(float)", "tensor(double)" },
+        "Constrain output types to float tensors.")
+    .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+        if (ctx.getAttribute("dtype") != nullptr) 
+          propagateElemTypeFromAttributeToOutput(ctx, "dtype", 0);
+        else
+          propagateElemTypeFromInputToOutput(ctx, 0, 0);
+        if (!hasNInputShapes(ctx, 1)) {
+          return;
+        }
+        propagateShapeFromInputToOutput(ctx, 0, 0);
+    });
+
+ONNX_OPERATOR_SCHEMA(Multinomial)
+    .SetDoc(R"DOC(
+Generate a tensor of samples from a multinomial distribution according to the probabilities
+of each of the possible outcomes.
+)DOC")
+    .Attr(
+        "sample_size",
+        "Number of times to sample.",
+        AttributeProto::INT,
+        static_cast<int64_t>(1))
+    .Attr(
+        "seed",
+        "(Optional) Seed to the random generator, if not specified we will auto generate one.",
+        AttributeProto::FLOAT,
+        OPTIONAL)
+    .Attr(
+        "dtype",
+        "(Optional) The data type for the elements of the output tensor, if not specified, we will use int32.",
+        AttributeProto::INT,
+        OPTIONAL)

```

Comment:
The default value can be specified in the schema here.

Reply:
fixed.